src/Fixer/Whitespace/MethodChainingIndentationFixer.php

```diff
@@ -0,0 +1,141 @@
+            array(new CodeSample("<?php\n\$user->setEmail('voff.web@gmail.com')\n         ->setPassword('233434');"),
+        ));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_OBJECT_OPERATOR);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+
+        for ($index = 1; $index < count($tokens); ++$index) {
+            if ($tokens[$index]->equals(array(T_OBJECT_OPERATOR))) {
+                if ($this->needLineBreak($index - 1, $tokens)) {
+                    $tokens[$index - 1]->setContent($tokens[$index - 1]->getContent().$lineEnding);
+                    --$index;
+                    continue;
+                }
+
+                $prev = $tokens[$index - 1];
+                $currentWhitespaces = $this->isLineBreak($prev);
+
+                if (false !== $currentWhitespaces) {
+                    $prevMeaningIndex = $tokens->getPrevMeaningfulToken($index);
+                    $rightWhitespaces = $this->getRightIndents($prevMeaningIndex, $tokens);
+
+                    if ($currentWhitespaces !== $rightWhitespaces) {
+                        $prev->setContent($lineEnding.$rightWhitespaces);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param int    $index
+     * @param Tokens $tokens
+     *
+     * @return string
+     */
+    private function getRightIndents($index, Tokens $tokens)
+    {
+        $indent = $this->whitespacesConfig->getIndent();
+
+        for ($i = $index; $i >= 0; --$i) {
+            $currentWhitespaces = $this->isLineBreak($tokens[$i]);
+
+            if (false !== $currentWhitespaces) {
+                if ($tokens[$i + 1]->equals(array(T_OBJECT_OPERATOR))) {
+                    return $currentWhitespaces;
+                }
+
+                return $currentWhitespaces.$indent;
+            }
+        }
+
+        return $indent;
+    }
+
+    /**
+     * @param int    $index
+     * @param Tokens $tokens
+     *
+     * @return bool
+     */
+    private function needLineBreak($index, Tokens $tokens)
+    {
+        $prevMeaningful = $tokens->getPrevMeaningfulToken($index);
+        $isComment = false;
+
+        for ($i = $index; $i > $prevMeaningful; --$i) {
+            if ($tokens[$i]->equals(array(T_OBJECT_OPERATOR)) || false !== $this->isLineBreak($tokens[$i])) {
+                return $isComment;
+            }
+
+            if ($tokens[$i]->equalsAny(array(array(T_COMMENT), array(T_DOC_COMMENT), array(T_START_HEREDOC)))) {
+                $isComment = true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param Token $token
+     *
+     * @return string|bool
+     */
+    private function isLineBreak(Token $token)
+    {
+        $matches = array();
```

Comment:
Initializing this variable is not required as the `preg_match()` call will do it.